R/tm_a_pca.R

```diff
@@ -269,10 +269,40 @@ srv_a_pca <- function(id, data, reporter, filter_panel_api, dat, plot_height, pl
       )
     }
 
-    anl_merged_input <- teal.transform::merge_expression_module(
+
+    rule_diff <- function(other) {
+      function(value) {
+        others <- selector_list()[[other]]()$select
+        if (isTRUE(is.element(value, others)))
+          "Response must not have been used for PCA."
+      }
+    }
+
+    selector_list <- teal.transform::data_extract_multiple_srv(
+      data_extract = list(dat = dat, response = response),
       datasets = data,
-      join_keys = get_join_keys(data),
-      data_extract = list(dat = dat, response = response)
+      select_validation_rule = list(
+        dat = ~if (length(.) < 2L) "Please select more than 1 variable to perform PCA.",
+        response = shinyvalidate::compose_rules(
+          shinyvalidate::sv_optional(),
+          rule_diff("dat")
+        )
+      )
+    )
+
+    rule_dupl <- function(value) {
```

Comment:
So you're not actually using value here, which although is ok, maybe isn't that clear?